src/applications/apps.js

```diff
@@ -457,7 +457,7 @@ export function toDynamicPathValidatorRegex(path) {
         // use charAt instead as we could not use es6 method endsWith
         regexStr.charAt(regexStr.length - 1) === "/"
           ? `${regexStr}.*$`
-          : `${regexStr}(\/.*)?$`;
+          : `${regexStr}([/#?].*)?$`;
```

Comment:
As far as I know, it is not possible for `?` to be present in the test string, since we're removing `location.search` from the URL before executing the regex. See code below:

https://github.com/single-spa/single-spa/blob/b870a1f4228e556ab92d5637f8f5b37931d2bf9e/src/applications/apps.js#L420

```suggestion
          : `${regexStr}([/#].*)?$`;
```

Reply:
Thanks, Joel, Sure, that makes sense, I'll make the required changes, also since you made me note `location.search` being removed, the line below I think could go as well:

https://github.com/single-spa/single-spa/blob/b870a1f4228e556ab92d5637f8f5b37931d2bf9e/src/applications/apps.js#L421

Check my next commit.